pennylane/beta/tapes/qubit_param_shift.py

```diff
@@ -0,0 +1,220 @@
+            params (list[Any]): the quantum tape operation parameters
+
+        Keyword Args:
+            shift (float): the parameter shift value
+
+        Returns:
+            array[float]: 1-dimensional array of length determined by the tape output
+            measurement statistics
+        """
+        op = self._par_info[idx]["op"]
+        p_idx = self._par_info[idx]["p_idx"]
+
+        s = (
+            np.pi / 2
+            if op.grad_recipe is None or op.grad_recipe[p_idx] is None
+            else op.grad_recipe[p_idx]
+        )
+        s = options.get("shift", s)
+
+        shift = np.zeros_like(params)
+        shift[idx] = s
+
+        shift_forward = np.array(self.execute_device(params + shift, device))
+        shift_backward = np.array(self.execute_device(params - shift, device))
+
+        return (shift_forward - shift_backward) / (2 * np.sin(s))
+
+    def parameter_shift_var(self, idx, device, params, **options):
+        r"""Partial derivative using the parameter-shift rule of a tape consisting of a mixture
+        of expectation values and variances of observables.
+
+        Args:
+            idx (int): trainable parameter index to differentiate with respect to
+            device (.Device, .QubitDevice): a PennyLane device
+                that can execute quantum operations and return measurement statistics
+            params (list[Any]): the quantum tape operation parameters
+
+        Returns:
+            array[float]: 1-dimensional array of length determined by the tape output
+            measurement statistics
+        """
+        # Temporarily convert all variance measurements on the tape into expectation values
+        for i in self.var_idx:
+            self._measurements[i].return_type = qml.operation.Expectation
+
+        # Get <A>, the expectation value of the tape with unshifted parameters. This is only
+        # calculated once, if `self._evA` is not None.
+        if self._evA is None:
+            self._evA = np.asarray(self.execute_device(params, device))
+
+        # evaluate the analytic derivative of <A>
+        pdA = self.parameter_shift(idx, device, params, **options)
+
+        # For involutory observables (A^2 = I) we have d<A^2>/dp = 0.
+        # Currently, the only observable we have in PL that may be non-involutory is qml.Hermitian
+        involutory = [i for i in self.var_idx if self.observables[i].name != "Hermitian"]
+
+        # If there are non-involutory observables A present, we must compute d<A^2>/dp.
+        non_involutory = set(self.var_idx) - set(involutory)
+        original = []
+
+        for i in non_involutory:
+            # We need to calculate d<A^2>/dp; to do so, we replace the
+            # involutory observables A in the queue with A^2.
+            original[:0] = [self._measurements[i]]
+            obs = self._measurements[i].obs
+
+            w = obs.wires
+            A = obs.matrix
+
+            new_obs = qml.Hermitian(A @ A, wires=w)

```

Comment:
We have two solutions:

* Simply add `qml.Hermitian(A @ A, wires=w, do_queue=False)`

* Place the observable within a dummy tape context:
  ```python
  with QuantumTape():
      new_obs = qml.Hermitian(A @ A, wires=w)
  ```
  to ensure it gets queued to this dummy context, rather than another one that might be present.

The second option has overhead associated with it (it will created an annotated queue, then process it, etc.). So in the interest of optimization, I will just go with `do_queue=False`. This also feels a bit cleaner, we avoid the queuing altogether.

Reply:
perhaps works for now, but I'd like to nuke `do_queue` forever